### PR TITLE
CTCTRADERS 2026- Transport Details models updated to reflect journey changes in User answers

### PR DESCRIPTION
**Pull request by sawan07**

### CODE REVIEW

File: app/models/journeyDomain/TransportDetails.scala [hunk 1]
```diff
@@ -54,7 +54,8 @@ object TransportDetails {
         UserAnswersReader[Mode5or7].widen[InlandMode] orElse
         UserAnswersReader[NonSpecialMode].widen[InlandMode]
 
-    case class Rail(code: Int) extends InlandMode
+    //    final case class Rail(code: Int) extends InlandMode
```
Comment (MatthewAlexanderBeck) [inline app/models/journeyDomain/TransportDetails.scala:57]: Can these be removed?

File: app/models/journeyDomain/TransportDetails.scala [hunk 2]
```diff
@@ -86,15 +99,20 @@ object TransportDetails {
...
+          .map(Mode5or7(_))
+      //      implicit val userAnswersReaderMode5or7: UserAnswersReader[Mode5or7] =
+      //        InlandModePage.reader
+      //          .map(_.toInt)
+      //          .filter(Mode5or7.Constants.codes.contains(_))
+      //          .flatMap(
+      //            code =>
+      //              NationalityAtDeparturePage.reader
+      //                .map(Mode5or7(code, _))
+      //          )
```
Comment (MatthewAlexanderBeck) [inline app/models/journeyDomain/TransportDetails.scala:111]: and here please

**Review by MatthewAlexanderBeck — APPROVED** [review]

File: app/models/journeyDomain/TransportDetails.scala (hunk 2 above)
Comment (sawan07) [reply app/models/journeyDomain/TransportDetails.scala]: done

File: app/models/journeyDomain/TransportDetails.scala (hunk 1 above)
Comment (sawan07) [reply app/models/journeyDomain/TransportDetails.scala:57]: done

**Pull request merged**